src/shapes/object.class.js

```diff
@@ -1002,6 +1004,9 @@
       else if (key === 'shadow' && value && !(value instanceof fabric.Shadow)) {
         value = new fabric.Shadow(value);
       }
+      else if (key === 'controls') {
+        value.attach(this);
```

Comment:
not sure we need this because I couldn't get `controls` to be set in `setOptions` without breaking all tests